vars/named_vars.go

```diff
@@ -0,0 +1,54 @@
+package vars
+
+import (
+	"fmt"
+	"strings"
+)
+
+type NamedVariables struct {
```

Comment:
This could actually be a type alias:

```go
type NamedVariables map[string]Variables
```

And then it can be constructed like `vars.NamedVariables{}` instead of constructing a map. And then `NewNamedVariables` could be removed.